packages/cli/src/handlers/download.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-await-in-loop */
+import {
+    ApiChartAsCodeListResponse,
+    ApiChartAsCodeUpsertResponse,
+    AuthorizationError,
+    SavedChart,
+} from '@lightdash/common';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { getConfig } from '../config';
+import GlobalState from '../globalState';
+import { checkLightdashVersion, lightdashApi } from './dbt/apiClient';
+
+const DOWNLOAD_FOLDER = 'target';
+export type DownloadHandlerOptions = {
+    verbose: boolean;
+};
+export const downloadHandler = async (
```

Comment:
Code for the CLI , it ony makes a single request to the API, and does file manipulation.